### ISSUE TITLE
Use ARC locally and QM options (Do I need to have Gaussian to run ARC?)

### ISSUE THREAD

**Issue by NUSCondy** [ReactionMechanismGenerator/ARC#450]
Hello,
I want to run ARC locally so I am really confused how to set the server info. Thus errors happened when I run it.
The default qm calculation software is Gaussian; however, it seems Gaussian is commercial software, is it already included in RMG? I remember mopac is included. If not, does that mean that I have to install Gaussian manually? Do you have any suggestions?
Looking forward to your reply!

**Comment by alongd** [comment]
Hi,

You can run ARC either from some computer (e.g., your laptop) so it communicates with a server, or have ARC installed on the server. We call the later "local" - where ARC is installed on the server you want it to communicate with.

In both cases, you will need to have an electronic structure software already installed on your server. ARC currently supports Gaussian, QChem, Molpro, Terachem, and Orca. Most of these are proprietary software, requiring a purchased license. Orca should be free of charge for academia users. Orca is famous for it's very nice implementation of DLPNO-CCSD(T) methods for high accuracy energy calculations of even relatively large molecules. However, I haven't tried DFT optimizations on Orca (it might be good, but I don't know). Anyway, ARC should support it. We are also currently working on adding support for Psi4, which is completely free (open source) - stay tuned.
In short, you can try running ARC with Orca if you don't have access to any of the other software and don't plan on purchasing a license.  Give it a shot, I'm happy to help.

**Comment by NUSCondy** [comment]
Hi @alongd ,
I have already installed orca on my computer with mpi available.
Actually, I am trying to run it on my own workshop so I don't need to submit the job.
So now how should I set it? And how can I specify an orca job in the inputs.py with a DLPNO CCSD(T) method and 7ZaP basis set?
Looking forward to your reply!

**Comment by alongd** [comment]
Currently ARC supports job submission through server cluster software: OGE, Slurm, PBS (and soon others). Did I understand correctly that you're not using a server? We haven't set it up to run jobs without such queue software. It is possible, and will require some work.

**Comment by NUSCondy** [comment]
Hello @alongd ，
Because I am using Unbuntu (WSL) on windows, I'm not sure if PBS job could be run on it. And do I have to do any settings if I wanna use orca?

**Comment by alongd** [comment]
So please check. You'll need a cluster software. I'm not sure how to do that on a Windows subsystem. It's probably possible, but outside of my domain. **After you do that**, you need to link Orca to the "server" in ARC's settings, see https://reactionmechanismgenerator.github.io/ARC/installation.html#associating-software-with-servers

**Comment by NUSCondy** [comment]
Hi @alongd ,
Now I have installed Slurm on my local workstation and run ARC with ORCA. Below error is reported:
ValueError: Missing Arkane atom energy corrections for LevelOfTheory(method='dlpnoccsd(t)',basis='def2tzvp',software='orca')
Does that mean I still miss something to run ARC?
And I am not familiar with how to set the level of theory of ORCA and default setting to seem to be of Gaussian? Do you have any suggestions for the setting? Does anyone in your group running ARC with ORCA?

**Comment by alongd** [comment]
Hi, glad to hear! You are right, there seem to be a mismatch: To compute Thermo data we use Arkane for the statistical mechanics, and it has a particular list of bond additivity corrections (BAC), see [here](http://reactionmechanismgenerator.github.io/RMG-Py/users/arkane/input.html#model-chemistry) and [here](https://github.com/ReactionMechanismGenerator/RMG-database/blob/master/input/quantum_corrections/data.py).
ARC may run without BAC by setting `bac_type` to `None`. This means we are not correcting for systematic (epistemic) sources of uncertainty. For now, try using this setting, we will soon add BAC values for some of Orca's methods, I'll make it a priority and update you.
You may still compute rate coefficients, though, which do not require BAC values (still set `bac_type` to `None`, but there's no down side to it)

**Comment by NUSCondy** [comment]
Hi @alongd ,
I find where the error happened in level.py and set bac_type to None:
    def to_arkane_level_of_theory(self,
                                  variant: Optional[str] = None,
                                  bac_type: None,
                                  comprehensive: bool = False,
                                  raise_error: bool = False,
                                  warn: bool = True,
                                  ) -> Optional[LevelOfTheory]:
But the error showed up:
Traceback (most recent call last):
  File "/mnt/c/WSL/ARC/ARC.py", line 12, in <module>
    from arc.common import read_yaml_file
  File "/mnt/c/WSL/ARC/arc/__init__.py", line 2, in <module>
    import arc.main
  File "/mnt/c/WSL/ARC/arc/main.py", line 38, in <module>
    from arc.level import Level
  File "/mnt/c/WSL/ARC/arc/level.py", line 306
    def to_arkane_level_of_theory(self,
                                 ^
SyntaxError: non-default argument follows default argument

**Comment by NUSCondy** [comment]
Hello, there is still something wrong with slurm or PBS install, that is, cannot connect to the subsystem. So I wonder how to run ARC without this submitting software. Like an rmg run. Where can I do some modifications that ARC could skip the job submission schedule?

**Comment by alongd** [comment]
If you want to have ARC execute an electronic structure software without a server and without a cluster management software you'll need to write a module under the job folder parallel to SSH and Local that executes jobs on your machine.
Job submission is at the heart of ARC, so that might involve modifications to other files as well.
If you don't have a server, my recommendation is to get a Linux system (don't work with a subsystem) and try install Slurm on it.

**Comment by alongd** [comment]
Can you upload the input file created by ARC?
(BTW, this is a separate issue, would be helpful to discuss it in a new Issue so it is searchable by future users)
